df_auth/viewsets.py

```diff
@@ -52,3 +70,40 @@ def blacklist(self, request, *args, **kwargs):
 class OTPViewSet(ValidationOnlyCreateViewSet):
     serializer_class = OTPObtainSerializer
     permission_classes = (permissions.AllowAny,)
+
+
+class SocialAuth(viewsets.GenericViewSet):
+    serializer_class = SocialAuthInputSerializer
+
+    def get_object(self):
+        user = self.request.user
+        self.request.backend.redirect_uri = settings.REST_SOCIAL_OAUTH_REDIRECT
+        user = user_is_authenticated(user) and user or None
+        self.request.backend.STATE_PARAMETER = False
+        user = self.request.backend.complete(user=user)
+        return user
+
+    @action(methods=["post"], detail=False)
+    def signin(self, request):
```

Comment:
why is it no possible to have this logic in serializer validate following the same pattern of the simplejwt?

Is it really necessary to override get_object? This does not feel intuitive.

Reply:
Not resolved yet. I'm still working on that 🙂

Reply:
Done, removed get_object function and used create functions as we create tokens.